### ISSUE TITLE
Make the Buttons more accessible

### ISSUE THREAD

**Issue by tkshill** [tkshill/Quarto#21]
# Issue Context
Currently the buttons for the gamepiece shapes are inaccessible. Their `label` attributes are set to the SVG images of the shapes themselves, rather than text that can be picked up by a screen reader. 

Current button implementation for board.
```
viewCellButton : Cell -> Element Msg
viewCellButton cell =
    Input.button
        [ Border.color Styles.blue, Border.width 5 ]
        { onPress = Just (PlaceAttempt cell)
        , label = viewCell cell
        }
```
Current button implementation for available gamepiece
```
viewRemainingPiecesButton : Gamepiece -> Element Msg
viewRemainingPiecesButton gamepiece =
    let
        svgImage =
            viewGamepiece gamepiece
    in
    Input.button [] { onPress = Just (Clicked gamepiece), label = svgImage }
```

# Suggested Solution
Refactor the way the view renders the available pieces and the game board to do the following
- Use the function suggested in the additional resources to modify the button attributes so they can have aria-label text. The text should be a string representation of the gamepiece. e.g. _"Square Colour1 Filled Small"_, _"Circle Colour2 Hollow Large"_
- Alternatively, the buttons can be change to have text based labelling (the same text as suggested for the aria-labels above), but maybe hide the labels so they can be picked up by a screen reader, but not seen by the average user.
- The shapes can be rendered as image backgrounds for the buttons, rather than labels? 
- A tooltip that appears when a mouse hovers over a shape and describes it would be neat

# Alternatives Considered
If it's not possible to add the aria-labels or use the shapes as backgrounds to the buttons themselves, the buttons may need to be placed into individual Els (divs) and then have those use the shapes as backgrounds.

# Additional Resources
Elm-UI has a module with some features that help with accessiblity. [This function](https://package.elm-lang.org/packages/mdgriffith/elm-ui/latest/Element-Region#description) looks like it can provide what we're looking for.

See [here](https://github.com/tkshill/Quarto/blob/main/README.md#run-and-install) in the read me for how to run and install the application.

See [here](https://github.com/tkshill/Quarto/blob/main/CONTRIBUTING.md#the-basics) in contributing for the basics on forking and cloning the repository.

**Comment by dominicduffin1** [comment]
I'd be interested in taking this on!

One question - The Region.description needs a `String` representation of the piece - the closest function I can find in the code produces a `List String` - does this need some work or am I missing something?

**Comment by tkshill** [comment]
Hey! Yeah, you can absolutely tackle this one.

I know we have a `gamepieceToList: Gamepiece -> List String` that makes a list of strings so I'd say maybe use that and transform to the string you want? We can talk about it more in depth in slack though.

Functions that immediately come to mind for me:
[concat](https://package.elm-lang.org/packages/elm/core/latest/String#concat)
[foldl](https://package.elm-lang.org/packages/elm/core/latest/List#foldl)
[foldl1](https://package.elm-lang.org/packages/elm-community/list-extra/latest/List-Extra#foldl1)

**Comment by dominicduffin1** [comment]
Thank you @tkshill - that sounds like a good plan - I'll read the docs for those functions!
